### ISSUE TITLE
Slow response times and 522 errors

### ISSUE THREAD

**Issue by mohanrao** [andrieslouw/imagesweserv#211]
Hi,

Today we observerd for some cases Images.weserv.nl is slow and sometimes returning 522 respone codes

**Comment by mohanrao** [comment]
@kleisauke seems like cloudfare issue. Just guessing. most of them are 522 response codes

**Comment by raRaRa** [comment]
https://www.cloudflarestatus.com/?_ga=2.49098022.1024970801.1580152244-f65b9b4d41bec4f33bfa4302bc2d1ac0

Are you located in Paris?

**Comment by andrieslouw** [comment]
Correct, [our servers are mainly around Paris](https://images.weserv.nl/faq/#where-are-your-servers-located).

Additionally, we had an issue with blockrules not propagating to Cloudflare. Due to this, users that should've been blocked by our rate limit, weren't filtered at the edge. This caused saturation of our CPU's, spikes in bandwidth, and spikes in response times, which yields 522 responses.

We added some servers now, and temporarily patched the rate limit issue. Sorry for the trouble caused..

**Comment by andrieslouw** [comment]
Please open a new issue if this ever happens again, I'll close this issue for now.
